### PR TITLE
Add support for passing local files to server as raw data

### LINKED ISSUE #96: Support passing local files with the native python client
## Description

Add support for passing local files as data, rather than as a reference. 

`emu.ncmeta(dataset=<path to file on disk>)` should read the file and pass it to the server in the request.

### PR DESCRIPTION
**Pull request by huard**
## Overview

This PR fixes #96 

Changes:

* Added support for path to local files ('/...' and 'file:///')
* Added support for file-like object (`BaseIO`)
* Added support for Path instances
* If server is localhost and Path to file is local, send reference
* Updated documentation and notebook

Test suite requires https://github.com/bird-house/emu/pull/74

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -70,6 +71,15 @@ def test_wps_client_multiple_output(wps):
     assert output.output2 == "19"
 
 
+@pytest.mark.online
+def test_wps_wordcounter(wps):
+    fn = '/tmp/text.txt'
+    with open(fn, 'w') as f:
+        f.write('Just an example')
+    out = wps.wordcounter(text=fn).get(asobj=True)
```
[the remark this answers is not in the document]
Comment (huard) [reply tests/test_client.py]: No, I'm using the file name afterward, so I need to make sure the file is written to disk and closed.

File: birdy/client/utils.py [hunk 2]
```diff
@@ -163,14 +164,51 @@ def format_type(obj):
     return doc
 
 
-def to_owslib(value, data_type):
+def as_raw(url, value):
```
[the remark this answers is not in the document]
Comment (huard) [reply birdy/client/utils.py]: is_embedded_in_request ?

**Comment by huard** [comment]
Does OWSLib know when to tag content with CDATA ? As long as we're passing trivial text and base64, we're good, but if not, that could become a problem.

**Comment by cehbrecht** [comment]
`owslib` does not automatically use cdata encoding for complex inputs:
https://github.com/geopython/OWSLib/blob/dcbc8db9a8eafa715a0469317bf23ea07de17326/owslib/wps.py#L1619

We may prepare this in birdy. But probably we would like to move the *encoding* part to `owslib` (base64, raw, cdata, ...). The encoding is decided by the mime-type.

**Comment by cehbrecht** [comment]
Just to reference comments in pywps PR:
https://github.com/geopython/pywps/pull/444

**Comment by cehbrecht** [comment]
... codacy complain belongs to this PR.
> Unused urljoin imported from six.moves.urllib.parse

**Comment by cehbrecht** [comment]
@huard fixed `urljoin` import and run successfully manual tests. Shall I merge?

**Pull request merged**